OpenLIFUSonicationPlanner/OpenLIFUSonicationPlanner.py

```diff
@@ -152,6 +171,94 @@ def setParameterNode(self, inputParameterNode: Optional[OpenLIFUSonicationPlanne
             # Note: in the .ui file, a Qt dynamic property called "SlicerParameterName" is set on each
             # ui element that needs connection.
             self._parameterNodeGuiTag = self._parameterNode.connectGui(self.ui)
+            self.addObserver(self._parameterNode, vtk.vtkCommand.ModifiedEvent, self.checkCanPlan)
+        self.checkCanPlan()
+    
+
+    def checkCanPlan(self, caller = None, event = None) -> None:
+
+        # If all the needed objects/nodes are loaded within the Slicer scene, all of the combo boxes will be enabled
+        # This means that the plan button can be enabled
+        if self.ui.TargetComboBox.enabled and self.ui.ProtocolComboBox.enabled and self.ui.VolumeComboBox.enabled and self.ui.TargetComboBox.enabled:
+            self.ui.PlanPushButton.enabled = True
+        else:
+            self.ui.PlanPushButton.enabled = False
```

Comment:
when the plan button becomes disabled can we set a tooltip text on it explaining why it is disabled / what the user would have to do to enable it? and unset that tooltip (or set it to something else that is descriptive of what pressing the button would do) when the button becomes enabled again